### ISSUE TITLE
CANNOT Reproduce your results: your results in the paper are absolutely suspectable

### ISSUE THREAD

**Issue by ghost** [Barabasi-Lab/AI-Bind#2]
Dear authors,

Sorry to disturb you, but there are several severe issues found in your repo:

1. cannot install the env based on your guidelines. For other readers, I found that using `python==3.8` works instead of `python==3.6.6`.
2. your provided data is either deleted, or not achievable.
3. your files in the `dockerfile` cannot be downloaded as well.
4. there are no concrete steps provided in this repo.

Based on these observations, we have reasonable clues to infer that your results and data in the paper are totally suspectable.

Please consider addressing these issues soon.

Best,

**Comment by ChatterjeeAyan** [comment]
Thanks for bringing the issues to our notice. 

1. This is due to the incompatibility of certain dependencies. You can try skipping the conflicting dependencies like contourpy and installing Matplotlib compatible with Python 3.6.6. While using Python 3.8, re-train the VecNet models.
Can you please try this docker image: https://hub.docker.com/r/omairs/foodome2? This should make the installation process smoother. 
2. The datafiles are available in Zenodo: https://zenodo.org/records/7226641. I do see that the Dropbox files have been removed. I will make the necessary changes regarding that. 
Training data and pre-trained VecNet models are available under: /data/sars-busters-consolidated/GitData/
3. Try using the docker mentioned in (1). The other docker without the datafiles is available here: https://hub.docker.com/r/omairs/foodome2_no_data. 
4. Please refer to the notebook descriptions provided under Code and Data -> Code. For running AI-Bind on your data, please refer to the frontend notebook: https://github.com/Barabasi-Lab/AI-Bind/blob/main/VecNet-User-Frontend.ipynb. 

Hope this information is helpful.

**Comment by madnessfish** [comment]
Thank you for the interesting work. Within the data files provided, which files include the negative sets generated by the 7-hop method? And which are the original experimental data?

I found files named "absolute_negatives_bindingDB" and folders named "VecNet_Unseen_Nodes" and "VecNet_Unseen_Targets". Whether the train/val/test sets in "VecNet_Unseen_Nodes" and "VecNet_Unseen_Targets" were already filled with 7-hop negatives?

Also, whether the 7-hop method is available (is it create_n_hop_negatives function in AI-bind.py)? which test data we should input for generating the 7-hop method.

Thank you in advance!

**Comment by ChatterjeeAyan** [comment]
>> I found files named "absolute_negatives_bindingDB".

This file contains the negative sampling from BindingDB obtained by thresholding (10^6nM) on kinetic constants.

>> folders named "VecNet_Unseen_Nodes" and "VecNet_Unseen_Targets". Whether the train/val/test sets in "VecNet_Unseen_Nodes" and "VecNet_Unseen_Targets" were already filled with 7-hop negatives?

That is correct, the negatives are included in these files, specified by InChiKey, target_aa_code, and labels (Y). If you choose the InChiKey-target_aa_code pairs with Y==0, you get the negative samples. 

>> Also, whether the 7-hop method is available (is it create_n_hop_negatives function in AI-bind.py)?

This is also correct. This function computes higher exponents of the adjacency matrix and then samples the negatives. This function creates DTI pairs that are up to max_hop hop-distance apart. 7-hop negatives are used in training and 11-15 hop negatives are used in validation and test. This function has been used to create the negatives in this notebook: https://github.com/Barabasi-Lab/AI-Bind/blob/main/VecNet/VecNet-Unseen_Nodes.ipynb. Check cells 10 and 28. 

 >> which test data we should input for generating the 7-hop method.

You need to input the training DTI network for this, which is available here: /data/sars-busters-consolidated/interactions/dataset_1.csv. You can follow the Jupyter notebook linked above to explore the process. 

Hope this helps!
